packages/discordx/src/decorators/classes/DSimpleCommand.ts

```diff
@@ -203,79 +204,83 @@ export class DSimpleCommand extends Method {
+          // only digits
+          const id = args[index]?.replace(/\D/g, "");
 
-        // Role | undefined
-        if (op.type === "ROLE") {
-          if (!id?.length) {
+          // undefined
+          if (!args[index]?.length) {
             return undefined;
           }
 
-          return command.message.guild?.roles.resolve(id);
-        }
+          // Boolean
+          if (op.type === "BOOLEAN") {
+            return Boolean(args[index]);
+          }
 
-        // GuildMember | User | undefined
-        if (op.type === "USER") {
-          if (!id?.length) {
-            return undefined;
+          // Number
+          if (op.type === "NUMBER" || op.type === "INTEGER") {
+            return Number(args[index]);
           }
 
-          if (command.message.channel.type === "DM") {
-            return command.message.client.user?.id === id
-              ? command.message.client.users.resolve(id)
-              : command.message.author;
+          // Channel | undefined
+          if (op.type === "CHANNEL") {
+            if (!id?.length) {
+              return undefined;
+            }
+            return command.message.guild?.channels
+              .fetch(id)
+              .catch((err) => err);
           }
 
-          return command.message.guild?.members.resolve(id);
-        }
+          // Role | undefined
+          if (op.type === "ROLE") {
+            if (!id?.length) {
+              return undefined;
+            }
 
-        // GuildMember | User | Role | undefined
-        if (op.type === "MENTIONABLE") {
-          if (!id?.length) {
-            return undefined;
+            return command.message.guild?.roles.fetch(id).catch((err) => err);
           }
 
-          if (command.message.channel.type === "DM") {
-            return command.message.client.user?.id === id
-              ? command.message.client.users.resolve(id)
-              : command.message.author;
+          // GuildMember | User | undefined
+          if (op.type === "USER") {
+            if (!id?.length) {
+              return undefined;
+            }
+
+            if (command.message.channel.type === "DM") {
+              return command.message.client.user?.id === id
+                ? command.message.client.users.fetch(id).catch((err) => err)
+                : command.message.author;
+            }
+
+            return command.message.guild?.members.fetch(id).catch((err) => err);
           }
 
-          return (
-            command.message.guild?.members.resolve(id) ??
-            command.message.guild?.roles.resolve(id)
-          );
-        }
+          // GuildMember | User | Role | undefined
+          if (op.type === "MENTIONABLE") {
+            if (!id?.length) {
+              return undefined;
+            }
+
+            if (command.message.channel.type === "DM") {
+              return command.message.client.user?.id === id
+                ? command.message.client.users.fetch(id).catch((err) => err)
```

Comment:
Shouldn't you be able to just return `command.message.client.user` here?

Reply:
Also, why are you making this check? The ID could be any user out there, not just the bot and the author...

Reply:
You can also give a completely different user without using a mention by giving an ID directly, in which case this will just return the author.

Reply:
ok so, what is your suggestion, return null, if it's not a ``bot`` or ``author``?

Reply:
If it's impossible to resolve a user otherwise in a DM, then handle it like the rest and throw an error.

Reply:
It does, but we should not do that, a developer can request argument type string and then do it manually in his handler.

Reply:
![image](https://user-images.githubusercontent.com/43092101/154636285-6198e270-c66a-4d92-a8a4-74b004f06769.png)

Reply:
Why should we not resolve?

Reply:
You can also just right-click on someone -> Copy ID and send that as arg

Reply:
you can do that, we resolve if user wants that, our resolver works for guild and dm, it's not just dm

Reply:
Yeah, so: Why not resolve in DM?

Reply:
we are, we are resolving bot id or author id, otherwise null

Reply:
I feel like we're going in circles now, but just for fun:
Why are we not resolving other IDs?

Reply:
cause total member in DM, are just 2 members, I am not sure if group is DM or not.

Reply:
So, again: Are we not able to resolve users outside of the DM? If not, we should return an error like in other places.

Reply:
I think you have a misunderstanding, we should take this to discord chat.